From d7e8ae0b7157005b0dd69a370241115e104c41f6 Mon Sep 17 00:00:00 2001
From: Makar Semyonov <m.semenov@tssltd.ru>
Date: Wed, 5 Nov 2025 17:16:51 +0800
Subject: [PATCH 49/49] cifs: prevent NULL pointer dereference in UTF16
 conversion

There can be a NULL pointer dereference bug here. NULL is passed to
__cifs_sfu_make_node without checks, which passes it unchecked to
cifs_strndup_to_utf16, which in turn passes it to
cifs_local_to_utf16_bytes where '*from' is dereferenced, causing a crash.

This patch adds a check for NULL 'src' in cifs_strndup_to_utf16 and
returns NULL early to prevent dereferencing NULL pointer.

Found by Linux Verification Center (linuxtesting.org) with SVACE

Signed-off-by: Makar Semyonov <m.semenov@tssltd.ru>
Cc: stable@vger.kernel.org
Signed-off-by: Steve French <stfrench@microsoft.com>
---
 fs/cifs/cifs_unicode.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/fs/cifs/cifs_unicode.c b/fs/cifs/cifs_unicode.c
index e7582dd79179..77d2109a59bd 100644
--- a/fs/cifs/cifs_unicode.c
+++ b/fs/cifs/cifs_unicode.c
@@ -618,6 +618,9 @@ cifs_strndup_to_utf16(const char *src, const int maxlen, int *utf16_len,
 {
 	int len;
 	__le16 *dst;
+
+	if (!src)
+		return NULL;
 
 	len = cifs_local_to_utf16_bytes(src, maxlen, cp);
 	len += 2; /* NULL */
-- 
2.43.0

